mm2src/mm2_main/src/lp_network.rs

```diff
@@ -138,36 +139,68 @@ async fn process_p2p_message(
     mut message: GossipsubMessage,
     i_am_relay: bool,
 ) {
+    fn is_valid(topics: &Vec<TopicHash>) -> bool {
```

Comment:
These I mostly concern on high level OOP languages where resource/performance things don't matter much. If we transform the codebase to this OOP(mostly) paradigms, we will get pretty decent impact on performance and resource usage. e.g, if we lose the control on "single responsibility" paradigm, %99 of the CPUs will miss the branch prediction. Can you show me an example system level projects written in Rust where this bussiness/domain layer pardigms were applied?

Reply:
This certain case is not connected to the branch prediction.

If you extracted this method you would get the same binary code, [you can check it up](https://godbolt.org/), therefore if you do this it won't hurt productivity at all. As I know calling method takes less than 1 nanosecond, but allocating memory takes more than 100 nanosecond (Alex Xu), that is why we can afford ourselves to not stuff `process_p2p_message` with everything and keep code clean.

Reply:
I ref to the reason you rely on, not "just" for this particular function, and I didn't mean keep everything in `process_p2p_message`.

I said if you apply these OOP(mostly) princibles to the low level language, you will not get the benefit of modern hardwares which means like cache missing, huge pointer chains, using heap stack unnecessarly, branch-prediction missing. Which means around ~%30 less performance. Parsing code doesn't only give you 1 nanosecond `calling` overhead.

Reply:
What I'm asking you to do is a trivial decomposition

![Peek 2023-04-19 19-41](https://user-images.githubusercontent.com/801971/233111958-e5fe6b90-c0a6-42a9-adb7-79c90b0411c0.gif)

Reply:
> What I'm asking you to do is a trivial decomposition

But for what? It's the commong language feature of the compiler that you can write your small/little code block as a local function just like closures. Which provides benefits like encapsulating relevant functionalities inside of a one function(ofc not all of the logic), to avoid having too many accessible functions which obviously makes codebase harder to focus and maintain.

--

and, again, the point where I made on the perf impact was on the applying OOP specific princibles on the system level projects.

Reply:
tbh for me having this as inner function makes it easier to focus and maintain, I leave the decision to the team @KomodoPlatform/mm2. Maybe I am wrong.

Reply:
Yeah, this argument has gone on for far too long.
First of all, I think **each of you** is right in your own way.
Next, 
(i) binary code is literally identical
(ii) this `is_valid` function is small (16 lines), size of `process_p2p_message` is 124 lines, including `is_valid`, which is still ok. For our project average normal size of functions is about 120-150 lines. if its about 300 then its too big and it needs to be divided into smaller fns (which was Artem recommendation about 300 lines and I agree with him)
(iii) In terms of code maintenance, from my experience our team always notices large chunks of code that need to be decomposed. Right now we are talking about a very small function that is used only in one place of program, so code support won't be a problem.

Given the enumerations above, I believe that **the creator of the PR** has the right to leave their own version of the code.

When similar arguments about non-critical parts in the code are repeated, we can refer to this rule if no consensus has been reached (in the discussion of the non-critical part of the program). Of course each situation is unique and will be considered separately.

Reply:
Thanks @rozhkovdmitrii, @ozkanonur, @laruh for the interesting discussion. I will be working on these fixes instead of @ozkanonur as he requested since he will not be available/off in the next few days.

> Given the enumerations above, I believe that the creator of the PR has the right to leave their own version of the code.

I agree with @laruh here. We also have similar inner functions throughout the code so this is within our code conventions, if this function's block of code was used here straight away https://github.com/KomodoPlatform/atomicDEX-API/blob/87cd284a4ce11c267801c02d2d031e30a9fe9e5b/mm2src/mm2_main/src/lp_network.rs#L165-L168 without making it a function, I think we would not be having this discussion. The function encapsulates a single purpose for the code so it seems valid to me, no pun intended :)

@rozhkovdmitrii if you still have a concern about this, please don't be discouraged to comment about it. My idea is to not focus on a trivial change as this that we can't all agree on.

Reply:
> Guys, I'm sorry but you're wrong ) Let me explain

I don't think there is a right and wrong approach here, it's up to preference really and different devs with different experiences will have different opinions on this. In such cases @rozhkovdmitrii we get everybody's opinion to reach an agreement and to not waste time in discussing a trivial case as this one. I think most agreed to leave this as is, so let's do that :)